src/commonMethods/commonFunctions.js

```diff
@@ -79,16 +173,318 @@ export const calcAvgFPS = function() {
+      throw new Error('Could not determine IP network of the server that belongs to: ' + ip)
+  }
+}
+
+/**
+ * This function is used to restart the framework
+ */
+export const restartFramework = function() {
+  return this.$sequence([
+    () => killProcess('WPEFramework'),
+    () => killProcess('WPEProcess'),
+    () => startFramework(),
+  ])
+}
 /**
  * This function performs below operations on WebKitBrowser Plugin
  *  - Deactivate
  *  - Activate
  *  - Resume
  */
-export const webKitBrowserOps = function() {
+export const webKitBrowserStartAndResume = function() {
   return this.$sequence([
     () => pluginDeactivate.call(this, constants.webKitBrowserPlugin),
     () => pluginActivate.call(this, constants.webKitBrowserPlugin),
     () => webKitBrowserActions.call(this, constants.resume),
   ])
 }
+
+/**
+ * This function is used to get Plugin Info
+ * @param plugin
+ * @returns {Promise<AxiosResponse<any>>}
+ */
+export const getPluginInfo = function(plugin) {
+  return this.$http
+    .get(`http://${constants.host}:80/Service/${plugin}`)
+    .then(result => result)
+    .catch(err => err)
+}
+
+/**
+ * This function is used to Suspend or Resume a plugin
+ * @param plugin
+ * @param action
+ * @returns {Promise<unknown>}
+ */
+export const suspendOrResumePlugin = function(plugin, action) {
+  return this.$http
+    .post(`http://${constants.host}:80/Service/${plugin}/${action}`)
+    .then(result => result)
+    .catch(err => err)
+}
+
+/**
+ * This function is used to get System Info of Device Ingo Plugin
+ * @param plugin
+ * @returns {Promise<AxiosResponse<any>>}
+ */
+export const getDeviceInfo = function() {
+  return this.$thunder.api.DeviceInfo.systeminfo()
+    .then(result => {
+      this.$data.write('systeminfo', result)
+    })
+    .catch(err => err)
+}
+
+/**
+ * This function is used to get esn Info of Netflix Plugin
+ * @returns {Promise<unknown>}
+ */
+export const getNetflixPluginEsnInfo = function() {
+  return this.$thunder.api.Netflix.esn()
+    .then(result => result)
+    .catch(err => err)
+}
+
+/**
+ * This function is used to suspend or resume Info of Netflix Plugin
+ * @param state
+ * @returns {Promise<unknown>}
+ */
+export const suspendOrResumeNetflixPlugin = function(state) {
+  return this.$thunder.api.Netflix.state(state)
+    .then(result => result)
+    .catch(err => err)
+}
+
+/**
+ * This function is used to get Addresses Info of Device Info Plugin
+ * @param plugin
+ * @returns {Promise<AxiosResponse<any>>}
+ */
+export const getAddressesInfo = function() {
+  return this.$thunder.api.DeviceInfo.addresses()
+    .then(result => {
+      this.$data.write('addressinfo', result)
+    })
+    .catch(err => err)
+}
+
+/**
+ * This function is used to get Monitor Info
+ * @param plugin
+ * @returns {Promise<AxiosResponse<any>>}
+ */
+export const getMonitorInfo = function() {
+  return this.$thunder.api.Monitor.status()
+    .then(result => {
+      this.$data.write('monitorinfo', result)
```

Comment:
Perhaps it would be a good idea to _also_ return the result value in this method (besides storing it in this.$data for convenience). Makes the method more flexible.